services/ui/ws/display.cc

```diff
@@ -361,6 +361,22 @@ void Display::OnCloseRequest() {
     window_tree->client()->RequestClose(window_id.id);
 }
 
+// TODO(msisov, tonikitoo): maybe combine calls to get window_tree,
+// server_window and window_id to duplicating the same code used above?
+void Display::OnWindowStateChanged(ui::mojom::ShowState new_state) {
+  if (!window_server_->IsInExternalWindowMode())
+    return;
+
+  WindowTree* window_tree = window_server_->GetTreeForExternalWindowMode();
+  WindowManagerDisplayRoot* display_root =
+      window_manager_display_root_map_.begin()->second;
+  ServerWindow* server_window =
+      display_root->window_manager_state()->GetWindowManagerRoot(root_window());
+  ClientWindowId window_id;
+  if (window_tree && window_tree->IsWindowKnown(server_window, &window_id))
```

Comment:
I believe no other code in display.cc calls window_tree->client() directly

it is ok for us to do that, but I would add a small comment (not a tODO)

// We are calling window_tree_client.cc directly from here for the sake not changing window_tree.cc for such a simple 
// call, adding less code as possible downstream.